webview-ui/src/components/settings/ApiOptions.tsx

```diff
@@ -123,7 +127,52 @@ const ApiOptions = ({
 		info: selectedModelInfo,
 	} = useSelectedModel(apiConfiguration)
 
-	const { data: routerModels, refetch: refetchRouterModels } = useRouterModels()
+	const { data: initialRouterModels } = useRouterModels()
+
+	const defaultRouterModels: RouterModels = useMemo(
+		() => ({
+			openrouter: emptyModelRecord,
+			requesty: emptyModelRecord,
+			glama: emptyModelRecord,
+			unbound: emptyModelRecord,
+			litellm: emptyModelRecord,
+		}),
+		[],
+	)
+
+	const [currentRouterModels, setCurrentRouterModels] = useState<RouterModels>(
+		initialRouterModels || defaultRouterModels,
+	)
+
+	useEffect(() => {
+		if (initialRouterModels) {
+			setCurrentRouterModels(initialRouterModels)
+		} else {
+			setCurrentRouterModels(defaultRouterModels)
+		}
+	}, [initialRouterModels, defaultRouterModels])
+
+	// Listen for specific provider model updates
+	useEffect(() => {
+		const handler = (event: MessageEvent<any>) => {
```

Comment:
Can we use `useEvent` from `react-use` for this? That seems to be what we do everywhere. Also, can we type this more strongly than `any`?